devtools-integration-tests/src/test/java/org/ballerina/devtools/cmd/BuildNewCommandTest.java

```diff
@@ -0,0 +1,165 @@
+        Assert.assertTrue(Files.exists(packageDir));
+
+        Process build = executeBuildCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<>(Collections.singletonList("project_name")), this.envVariables);
+
+        Assert.assertTrue(packageDir.resolve("target").resolve("bin").resolve("project_name.jar").toFile().exists());
+    }
+
+    @Test(description = "Build package created from new command with main template")
+    public void testCompilingNewCommandMainTempProject() throws IOException, InterruptedException {
+        Process newCommand = executeNewCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<String>(Arrays.asList("main_sample", "-t", "main")), this.envVariables);
+
+        Path packageDir = this.tempWorkspaceDirectory.resolve("main_sample");
+        Assert.assertTrue(Files.exists(packageDir));
+
+        Process build = executeBuildCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<>(Collections.singletonList("main_sample")), this.envVariables);
+
+        Assert.assertTrue(packageDir.resolve("target").resolve("bin").resolve("main_sample.jar").toFile().exists());
+    }
+
+    @Test(description = "Build package created from new command with service template")
+    public void testCompilingNewCommandServiceTempProject() throws IOException, InterruptedException {
+        Process newCommand = executeNewCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<String>(Arrays.asList("service_sample", "-t", "service")), this.envVariables);
+
+        Path packageDir = this.tempWorkspaceDirectory.resolve("service_sample");
+        Assert.assertTrue(Files.exists(packageDir));
+
+        Process build = executeBuildCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<>(Collections.singletonList("service_sample")), this.envVariables);
+
+        Assert.assertTrue(packageDir.resolve("target").resolve("bin").resolve("service_sample.jar").toFile().exists());
+    }
+
+    @Test(description = "Build package created from new command with service template")
+    public void testCompilingNewCommandLibTempProject() throws IOException, InterruptedException {
+        Process newCommand = executeNewCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<String>(Arrays.asList("lib_sample", "-t", "lib")), this.envVariables);
+
+        Path packageDir = this.tempWorkspaceDirectory.resolve("lib_sample");
+        Assert.assertTrue(Files.exists(packageDir));
+
+        Process build = executeBuildCommand(DISTRIBUTION_FILE_NAME, this.tempWorkspaceDirectory,
+                new LinkedList<>(Collections.singletonList("lib_sample")), this.envVariables);
+
+        Assert.assertTrue(packageDir.resolve("target").resolve("bin").resolve("lib_sample.jar").toFile().exists());
```

Comment:
You can use the function `getExecutableJarPath()` here.
```suggestion
        Assert.assertTrue(getExecutableJarPath(packageDir, "lib_sample").toFile().exists());
```